### DISCUSSION TITLE
HA + RaspberryMatic Add-on + HM-MOD-RPI-PCB = Keine GPIO gefunden

### DISCUSSION THREAD

**Discussion by carusoe** [OpenCCU/OpenCCU#1725] [🇩🇪 Fragen&Antworten]
Hallo,

nachdem Ihr mir so viele Tipps gegeben habt zu dem HomeMatic USB-Sticks (https://github.com/jens-maus/RaspberryMatic/discussions/1651) habe ich auf meinem Raspi 4B mit 4GB Hassio installiert und bin nach der Anleitung von Jens zur Installation des RaspberryMatics Add-On vorgegangen. Zumindest war das der Plan. Denn ich bekomme keinen Kontakt zu meinen Sensoren. Nach einiger Zeit habe ich in dem RaspberryMatic-Add-On-Log folgende Zeilen gefunden:
Identifying Homematic RF-Hardware: ...HmRF: none, HmIP: none, OK
Updating Homematic RF-Hardware: no GPIO/USB connected RF-hardware found

Dann habe ich als erstes die Config.txt kontrolliert. Dort sind die entsprechenden Zeilen nicht mehr auskommentiert:
# Uncomment this to enable GPIO support for RPI-RF-MOD/HM-MOD-RPI-PCB
enable_uart=1
dtparam=i2c_arm=on
dtoverlay=miniuart-bt
dtoverlay=rpi-rf-mod

Ich bin über die Konsole per angeschlossener Tastatur in das Verzeichnis /mnt/boot/ gegangen und habe die config.txt geändert. 

Ich bekomme trotzdem HA nicht dazu meinen HM-MOD-RPI-PCB zu erkennen. Was mache ich falsch?

Viele Grüße

**Comment by bgaworsk** [comment]
Moin,

hatte das selbe Problem. Habe gerade in einem parallelen Thread den Hinweis gefunden, der für mich funktioniert hat.

In den Einstellungen des Add-ons den ```Protection Mode``` ausschalten:

<img width="1069" alt="image" src="https://user-images.githubusercontent.com/1122375/155019826-555e9b01-4b33-443e-8961-6081b3278eb5.png">

Letzter toggle switch im Screenshot. Danach ein Klick auf den Restart Button und siehe da:

```
Identifying Homematic RF-Hardware: ....HmRF: RPI-RF-MOD/GPIO, HmIP: RPI-RF-MOD/GPIO, OK
Updating Homematic RF-Hardware: RPI-RF-MOD: 4.4.12, OK
```

**Reply by ghost** [reply]
hi, bei mir gibt es diesen button für den protection mode nicht. ich habe raspberry Os mit Home assistant supervised. hat jemand einen tipp?

**Reply by jens-maus** [reply]
Das ist ja auch ein alter Beitrag hier (Januar 2022) und den Protection mode kann und muss man schon seit geraumer Zeit nicht mehr Abschalten im RaspberryMatic HA Addon. Deine Probleme die du vmtl. hast kommen woanders her, eben aus dem Punkt das du "Home Assistant Supervised" einsetzt und das mitunter bekannt dafür ist gerade bei direkten Hardwarezugriffen problematisch zu sein. Lies dir dazu genauer den Abschnitt in der offiziellen RaspberryMatic Doku dazu durch:

https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#using-homeassistant-supervised

**Reply by ghost** [reply]
Das habe ich alles befolgt Schritt für Schritt. Leider hat es mehrere Male nicht funktioniert

**Reply by jens-maus** [reply]
Tja, sagte ja: HA Supervised installation sind bei direkten Hardwarezugriffen "suboptimal". Nutz "HomeAssisant OS" und die Probleme (welche es auch immer sind) sollten weg sein.

**Reply by ghost** [reply]
Das läuft auf dem raspberry pi 3 leider sehr wenig stabil und die cpu wird sehr heiß. Ich hätte gerne einen gesteuerten Lüfter eingebaut zum kühlen, aber die gpio Steuerung ist bei haos ja leider nicht erreichbar (rpi rf mod belegt ja auch alle gpio Pins)

**Reply by Baxxy13** [reply]
Von HA-Supervised kann ich im Zusammenhang mit USB-Funklösungen auch nur abraten (GPIO nicht getestet). 
Da geht aktuell gar nichts.

Beste Lösung: 2 Pi's, einen für RM einen für HA. (weil ich auch vom RM-HA-AddOn abraten muss)
